integration-tests/soak/ocr_test.go

```diff
@@ -22,58 +30,183 @@ func TestOCRSoak(t *testing.T) {
 	// fmt.Println("---------------------")
 	config, err := tc.GetConfig("Soak", tc.OCR)
 	require.NoError(t, err, "Error getting config")
-	runOCRSoakTest(t, config, customNetworkTOML)
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, customNetworkTOML)
 }
 
 func TestOCRSoak_GethReorgBelowFinality_FinalityTagDisabled(t *testing.T) {
 	config, err := tc.GetConfig(t.Name(), tc.OCR)
 	require.NoError(t, err, "Error getting config")
-	runOCRSoakTest(t, config, "")
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, "")
 }
 
 func TestOCRSoak_GethReorgBelowFinality_FinalityTagEnabled(t *testing.T) {
 	config, err := tc.GetConfig(t.Name(), tc.OCR)
 	require.NoError(t, err, "Error getting config")
-	runOCRSoakTest(t, config, "")
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, "")
 }
 
 func TestOCRSoak_GasSpike(t *testing.T) {
 	config, err := tc.GetConfig(t.Name(), tc.OCR)
 	require.NoError(t, err, "Error getting config")
-	runOCRSoakTest(t, config, "")
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, "")
 }
 
 // TestOCRSoak_ChangeBlockGasLimit changes next block gas limit and sets it to percentage of last gasUsed in previous block creating congestion
 func TestOCRSoak_ChangeBlockGasLimit(t *testing.T) {
 	config, err := tc.GetConfig(t.Name(), tc.OCR)
 	require.NoError(t, err, "Error getting config")
-	runOCRSoakTest(t, config, "")
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, "")
+}
+
+// TestOCRSoak_RPCDownForAllCLNodes simulates a network chaos by bringing down network to RPC node for all Chainlink Nodes
+func TestOCRSoak_RPCDownForAllCLNodes(t *testing.T) {
+	config, err := tc.GetConfig(t.Name(), tc.OCR)
+	require.NoError(t, err, "Error getting config")
+	require.True(t, config.Network.IsSimulatedGethSelected(), "This test requires simulated geth")
+
+	namespace := fmt.Sprintf("%s-%s", "soak-ocr-simulated-geth", uuid.NewString()[0:5])
+	chaos, err := gethNetworkDownChaos(GethNetworkDownChaosOpts{
+		DelayCreate: time.Minute * 2,
+		Duration:    time.Minute * 5,
```

Comment:
we are sure we don't want these values to be configurable via TOML?

Reply:
@Tofel 

I had the same idea at the beginning (see my previous commits). But, now I see more downsides like:

1. We would need to create custom fields or structs for this in TOML
2. We would probably need to put other params as well like chaos description, chaos selectors, chaos mode, etc. (see chaos definition in tests). Do you think it it is useful for anyone to expose it in TOML?

Reply:
Hard to say at this point how people will use this, so maybe let's leave as-is until someone requests it?

Reply:
@Tofel yeah, I think this approach makes sense. It's very easy to update if needed. It's a "two-way door" decision, changeable and reversible according to Bezos framework that I really like https://coda.io/@hales/jeff-bezos-make-fast-decisions